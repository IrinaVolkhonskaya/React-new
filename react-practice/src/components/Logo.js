import React from 'react';
import logo from '../images/logo.png';

const Logo = () => (
  <img className="logo" src={logo} alt="logo cafe" width="100" height="70" />
);

export default Logo;
