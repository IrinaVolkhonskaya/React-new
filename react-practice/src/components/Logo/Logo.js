import React from "react";
import logo from "../AppHeader/assets/logo.png";

const Logo = ({ width, height}) => (
  <img
    className="logo"
    src={logo}
    alt="logo cafe"
    width={width}
    height={height}
  />
);

export default Logo;
