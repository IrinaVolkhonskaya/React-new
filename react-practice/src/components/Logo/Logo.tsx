import React from "react";
import logo from "../AppHeader/assets/logo.png";

interface ILogo {
  width: number,
  height: number,
}

const Logo: React.FC<ILogo> = ({ width, height}) => (
  <img
    className="logo"
    src={logo}
    alt="logo cafe"
    width={width}
    height={height}
  />
);

export default Logo;
