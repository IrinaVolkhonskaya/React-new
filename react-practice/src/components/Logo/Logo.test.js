import React from "react";
import Logo from "./Logo";

import { shallow } from "enzyme";
import Enzyme from "enzyme";
import Adapter from "enzyme-adapter-react-16";

Enzyme.configure({ adapter: new Adapter() });

describe("<Logo />", () => {
  it("should renders <Logo /> component", () => {
    const wrapper = shallow(<img className="logo" />);
    expect(wrapper.contains(<img className="logo" />)).toEqual(true);
  });

  it("should renders <Logo /> component with props", () => {
    const wrapper = shallow(<Logo width="100px" height="100px" />);
    expect(wrapper).toMatchSnapshot();
  });

  it("should renders <Logo /> component without props", () => {
    const wrapper = shallow(<Logo />);
    expect(wrapper).toMatchSnapshot();
  });
});
