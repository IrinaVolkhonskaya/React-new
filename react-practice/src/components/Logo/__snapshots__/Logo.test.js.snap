// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Logo /> should renders <Logo /> component with props 1`] = `
<img
  alt="logo cafe"
  className="logo"
  height="100px"
  src="logo.png"
  width="100px"
/>
`;

exports[`<Logo /> should renders <Logo /> component without props 1`] = `
<img
  alt="logo cafe"
  className="logo"
  src="logo.png"
/>
`;
